### PR TITLE
CASMPET-4660

### PR DESCRIPTION
**Pull request by leliasen-hpe**
Added goss test and script to automate check for postgres pod backups.

## Summary and Scope

This is a new automation feature. It is a goss test that ensures that recent PostgreSQL pod backups exist for specific clusters.

Change a new feature, it is backwards compatible.

## Issues and Related PRs

* Part of the 'Automation: Validate CSM Health (PET)' epic. CASM-2592

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * wasp
  * Hela
 
### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

This is a read only script that will report PASS or FAIL status.

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: goss-testing/scripts/postgresql_backups_check.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+print_results=0
+while getopts ph stack
+do
+    case "${stack}" in
+          p) print_results=1;;
+          h) echo "usage: postgres_backups_check.sh           # Only print 'PASS' upon success"
```
Comment (brantk-hpe) [inline goss-testing/scripts/postgresql_backups_check.sh:8]: The name of the script is postgresql_backups_check.sh

**Review by brantk-hpe — CHANGES_REQUESTED** [review]

**Review by bklei — DISMISSED** [review]

File: goss-testing/scripts/postgresql_backups_check.sh (hunk 1 above)
Comment (leliasen-hpe) [reply goss-testing/scripts/postgresql_backups_check.sh]: fixed

**Review by rgross727 — DISMISSED** [review]

**Review by brantk-hpe — DISMISSED** [review]

**Review by kimjensen-hpe — CHANGES_REQUESTED** [review]
goss-testing/tests/ncn/goss-k8s-postgres-backups.sh should be a .yaml file

**Comment by kimjensen-hpe** [comment]
> goss-testing/tests/ncn/goss-k8s-postgres-backups.sh should be a .yaml file

I fixed this but also unintentionally pulled in another recent change when I rebased master to avoid merge conflicts

**Review by kimjensen-hpe — APPROVED** [review]

**Review by rgross727 — APPROVED** [review]

**Review by bklei — APPROVED** [review]

**Comment by kimjensen-hpe** [comment]
rebase of this branch with master had issues as did the backport - will close this and opening a new PR based off a new/clean branch

**Pull request closed without merging**